<h1>Music Recommendation</h1>
<h2>Sumanth</h2>

<h3>Importing Libraries</h3>

In [80]:
import numpy as np
import pandas as pd
import csv
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
import lightgbm as lgb 
from tqdm import tqdm

<h2>Loading Data</h2>

In [81]:
ntr=7000
nts=3000

In [82]:
train=pd.read_csv("D:\\train.csv",nrows=ntr)

In [83]:
names=['msno','song_id','source_system_tab','source_screen_name',\
    'source_type','target']
test1=pd.read_csv('D:\\train.csv',names=names,skiprows=ntr,nrows=nts)
songs=pd.read_csv('D:\\songs.csv')
members=pd.read_csv('D:\\members.csv')

<h2>Analysing the Data</h2>

In [84]:
train.head()

msno  \
0  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   
1  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
2  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
3  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
4  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   

                                        song_id source_system_tab  \
0  BBzumQNXUHKdEBOB7mAJuzok+IJA1c2Ryg/yzTF6tik=           explore   
1  bhp/MpSNoqoxOIB+/l8WPqu6jldth4DIpCm3ayXnJqM=        my library   
2  JNWfrrC7zNN7BdMpsISKa4Mw+xVJYNnxXh3/Epw7QgY=        my library   
3  2A87tzfnJTSWqD7gIZHisolhe4DMdzkbd6LzO1KHjNs=        my library   
4  3qm6XTZ6MOCU11x8FIVbAGH5l5uMkT3/ZalWG1oo2Gc=           explore   

    source_screen_name      source_type  target  
0              Explore  online-playlist       1  
1  Local playlist more   local-playlist       1  
2  Local playlist more   local-playlist       1  
3  Local playlist more   local-playlist       1  
4              Explore  online-playlist       1

In [85]:
test1.head()

msno  \
0  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
1  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
2  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
3  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   
4  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   

                                        song_id source_system_tab  \
0  gpioqmDgSWfpkkCSQts5GMAGaM1TFH5zpVawYOjxIog=        my library   
1  CQ8ouYUSxeEMbNaql4pAL35CqqJKgd0Ms7tH3DmHOJ8=        my library   
2  NA/ZT/ard6XUWrnCvqbEnQLiXnYyP2m4Zq3jWvjLCPY=        my library   
3  wloUHWvwV3EYBJsb0hUG1tIeRlgkrf8QzkootXtY/v0=            search   
4  UZ75pCG2butQs9dVI1s4tedeDZGRwQ551HGdRsVm29Y=            search   

     source_screen_name      source_type  target  
0   Local playlist more    local-library       1  
1   Local playlist more    local-library       1  
2   Local playlist more    local-library       1  
3  Online playlist more  online-playlist       0  
4  Online playlist more  online-playlist       0

In [86]:
songs.head()

song_id  song_length genre_ids  \
0  CXoTN1eb7AI+DntdU1vbcwGRV4SCIDxZu+YD8JP8r4E=       247640       465   
1  o0kFgae9QtnYgRkVPqLJwa05zIhRlUjfF7O1tDw0ZDU=       197328       444   
2  DwVvVurfpuz+XPuFvucclVQEyPqcpUkHR0ne1RQzPs0=       231781       465   
3  dKMBWoZyScdxSkihKG+Vf47nc18N9q4m58+b4e7dSSE=       273554       465   
4  W3bqWd3T+VeHFzHAUfARgW9AvVRaF4N5Yzm4Mr6Eo/o=       140329       726   

        artist_name                            composer     lyricist  language  
0  張信哲 (Jeff Chang)                                  董貞          何啟弘       3.0  
1         BLACKPINK  TEDDY|  FUTURE BOUNCE|  Bekuh BOOM        TEDDY      31.0  
2      SUPER JUNIOR                                 NaN          NaN      31.0  
3             S.H.E                                 湯小康          徐世珍       3.0  
4              貴族精選                         Traditional  Traditional      52.0

In [87]:
members.head()

msno  city  bd gender  \
0  XQxgAYj3klVKjR3oxPPXYYFp4soD4TuBghkhMTD4oTw=     1   0    NaN   
1  UizsfmJb9mV54qE9hCYyU07Va97c0lCRLEQX3ae+ztM=     1   0    NaN   
2  D8nEhsIOBSoE6VthTaqDX8U6lqjJ7dLdr72mOyLya2A=     1   0    NaN   
3  mCuD+tZ1hERA/o5GPqk38e041J8ZsBaLcu7nGoIIvhI=     1   0    NaN   
4  q4HRBfVSssAFS9iRfxWrohxuk9kCYMKjHOEagUMV6rQ=     1   0    NaN   

   registered_via  registration_init_time  expiration_date  
0               7                20110820         20170920  
1               7                20150628         20170622  
2               4                20160411         20170712  
3               9                20150906         20150907  
4               4                20170126         20170613

<h2>Data Processing</h2>

In [88]:
test=test1.drop(['target'],axis=1)
ytr=np.array(test1['target'])

In [89]:
test.head()

msno  \
0  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
1  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
2  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
3  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   
4  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   

                                        song_id source_system_tab  \
0  gpioqmDgSWfpkkCSQts5GMAGaM1TFH5zpVawYOjxIog=        my library   
1  CQ8ouYUSxeEMbNaql4pAL35CqqJKgd0Ms7tH3DmHOJ8=        my library   
2  NA/ZT/ard6XUWrnCvqbEnQLiXnYyP2m4Zq3jWvjLCPY=        my library   
3  wloUHWvwV3EYBJsb0hUG1tIeRlgkrf8QzkootXtY/v0=            search   
4  UZ75pCG2butQs9dVI1s4tedeDZGRwQ551HGdRsVm29Y=            search   

     source_screen_name      source_type  
0   Local playlist more    local-library  
1   Local playlist more    local-library  
2   Local playlist more    local-library  
3  Online playlist more  online-playlist  
4  Online playlist more  online-playlist

<h2>Rearranging column to fit data into code </h2>

In [90]:
test_name=['id','msno','song_id','source_system_tab','source_screen_name','source_type']
test['id']=np.arange(nts)
test=test[test_name]

In [91]:
test.head()

id                                          msno  \
0   0  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
1   1  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
2   2  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
3   3  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   
4   4  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   

                                        song_id source_system_tab  \
0  gpioqmDgSWfpkkCSQts5GMAGaM1TFH5zpVawYOjxIog=        my library   
1  CQ8ouYUSxeEMbNaql4pAL35CqqJKgd0Ms7tH3DmHOJ8=        my library   
2  NA/ZT/ard6XUWrnCvqbEnQLiXnYyP2m4Zq3jWvjLCPY=        my library   
3  wloUHWvwV3EYBJsb0hUG1tIeRlgkrf8QzkootXtY/v0=            search   
4  UZ75pCG2butQs9dVI1s4tedeDZGRwQ551HGdRsVm29Y=            search   

     source_screen_name      source_type  
0   Local playlist more    local-library  
1   Local playlist more    local-library  
2   Local playlist more    local-library  
3  Online playlist more  online-playlist  
4  Online playlist more  online-playlist

In [92]:
songs.head()

song_id  song_length genre_ids  \
0  CXoTN1eb7AI+DntdU1vbcwGRV4SCIDxZu+YD8JP8r4E=       247640       465   
1  o0kFgae9QtnYgRkVPqLJwa05zIhRlUjfF7O1tDw0ZDU=       197328       444   
2  DwVvVurfpuz+XPuFvucclVQEyPqcpUkHR0ne1RQzPs0=       231781       465   
3  dKMBWoZyScdxSkihKG+Vf47nc18N9q4m58+b4e7dSSE=       273554       465   
4  W3bqWd3T+VeHFzHAUfARgW9AvVRaF4N5Yzm4Mr6Eo/o=       140329       726   

        artist_name                            composer     lyricist  language  
0  張信哲 (Jeff Chang)                                  董貞          何啟弘       3.0  
1         BLACKPINK  TEDDY|  FUTURE BOUNCE|  Bekuh BOOM        TEDDY      31.0  
2      SUPER JUNIOR                                 NaN          NaN      31.0  
3             S.H.E                                 湯小康          徐世珍       3.0  
4              貴族精選                         Traditional  Traditional      52.0

In [93]:
print('Data preprocessing...')
song_cols=['song_id','artist_name','genre_ids','song_length','language']
train=train.merge(songs[song_cols],on='song_id',how='left')
test=test.merge(songs[song_cols],on='song_id',how='left')

Data preprocessing...


<h2>Splitting into year, month and date</h2>

In [94]:
members.head()

msno  city  bd gender  \
0  XQxgAYj3klVKjR3oxPPXYYFp4soD4TuBghkhMTD4oTw=     1   0    NaN   
1  UizsfmJb9mV54qE9hCYyU07Va97c0lCRLEQX3ae+ztM=     1   0    NaN   
2  D8nEhsIOBSoE6VthTaqDX8U6lqjJ7dLdr72mOyLya2A=     1   0    NaN   
3  mCuD+tZ1hERA/o5GPqk38e041J8ZsBaLcu7nGoIIvhI=     1   0    NaN   
4  q4HRBfVSssAFS9iRfxWrohxuk9kCYMKjHOEagUMV6rQ=     1   0    NaN   

   registered_via  registration_init_time  expiration_date  
0               7                20110820         20170920  
1               7                20150628         20170622  
2               4                20160411         20170712  
3               9                20150906         20150907  
4               4                20170126         20170613

<h2>Dropping Registration time column</h2>

In [95]:
members['registration_year'] = members['registration_init_time'].apply(lambda x: int(str(x)[0:4]))
members['registration_month'] = members['registration_init_time'].apply(lambda x: int(str(x)[4:6]))
members['registration_date'] = members['registration_init_time'].apply(lambda x: int(str(x)[6:8]))

In [96]:
members.head()

msno  city  bd gender  \
0  XQxgAYj3klVKjR3oxPPXYYFp4soD4TuBghkhMTD4oTw=     1   0    NaN   
1  UizsfmJb9mV54qE9hCYyU07Va97c0lCRLEQX3ae+ztM=     1   0    NaN   
2  D8nEhsIOBSoE6VthTaqDX8U6lqjJ7dLdr72mOyLya2A=     1   0    NaN   
3  mCuD+tZ1hERA/o5GPqk38e041J8ZsBaLcu7nGoIIvhI=     1   0    NaN   
4  q4HRBfVSssAFS9iRfxWrohxuk9kCYMKjHOEagUMV6rQ=     1   0    NaN   

   registered_via  registration_init_time  expiration_date  registration_year  \
0               7                20110820         20170920               2011   
1               7                20150628         20170622               2015   
2               4                20160411         20170712               2016   
3               9                20150906         20150907               2015   
4               4                20170126         20170613               2017   

   registration_month  registration_date  
0                   8                 20  
1                   6                 28  
2                   4                 11  
3                   9                  6  
4                   1                 26

<h2>Left join with training and testing data</h2>

In [97]:
members_cols=members.columns
train = train.merge(members[members_cols], on='msno', how='left')
test = test.merge(members[members_cols], on='msno', how='left')

In [98]:
train = train.fillna(-1)
test = test.fillna(-1)

In [99]:
train.head()

msno  \
0  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   
1  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
2  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
3  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
4  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   

                                        song_id source_system_tab  \
0  BBzumQNXUHKdEBOB7mAJuzok+IJA1c2Ryg/yzTF6tik=           explore   
1  bhp/MpSNoqoxOIB+/l8WPqu6jldth4DIpCm3ayXnJqM=        my library   
2  JNWfrrC7zNN7BdMpsISKa4Mw+xVJYNnxXh3/Epw7QgY=        my library   
3  2A87tzfnJTSWqD7gIZHisolhe4DMdzkbd6LzO1KHjNs=        my library   
4  3qm6XTZ6MOCU11x8FIVbAGH5l5uMkT3/ZalWG1oo2Gc=           explore   

    source_screen_name      source_type  target      artist_name genre_ids  \
0              Explore  online-playlist       1         Bastille       359   
1  Local playlist more   local-playlist       1  Various Artists      1259   
2  Local playlist more   local-playlist       1              Nas      1259   
3  Local playlist more   local-playlist       1         Soundway      1019   
4              Explore  online-playlist       1      Brett Young      1011   

   song_length  language  city  bd  gender  registered_via  \
0       206471      52.0     1   0      -1               7   
1       284584      52.0    13  24  female               9   
2       225396      52.0    13  24  female               9   
3       255512      -1.0    13  24  female               9   
4       187802      52.0     1   0      -1               7   

   registration_init_time  expiration_date  registration_year  \
0                20120102         20171005               2012   
1                20110525         20170911               2011   
2                20110525         20170911               2011   
3                20110525         20170911               2011   
4                20120102         20171005               2012   

   registration_month  registration_date  
0                   1                  2  
1                   5                 25  
2                   5                 25  
3                   5                 25  
4                   1                  2

In [100]:
test.head()

id                                          msno  \
0   0  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
1   1  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
2   2  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
3   3  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   
4   4  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   

                                        song_id source_system_tab  \
0  gpioqmDgSWfpkkCSQts5GMAGaM1TFH5zpVawYOjxIog=        my library   
1  CQ8ouYUSxeEMbNaql4pAL35CqqJKgd0Ms7tH3DmHOJ8=        my library   
2  NA/ZT/ard6XUWrnCvqbEnQLiXnYyP2m4Zq3jWvjLCPY=        my library   
3  wloUHWvwV3EYBJsb0hUG1tIeRlgkrf8QzkootXtY/v0=            search   
4  UZ75pCG2butQs9dVI1s4tedeDZGRwQ551HGdRsVm29Y=            search   

     source_screen_name      source_type       artist_name genre_ids  \
0   Local playlist more    local-library     謝和弦 (R-chord)       465   
1   Local playlist more    local-library     謝和弦 (R-chord)       465   
2   Local playlist more    local-library  王菀之 (Ivana Wong)       458   
3  Online playlist more  online-playlist     Thomas Newman       921   
4  Online playlist more  online-playlist    Marcelo Zarvos       921   

   song_length  language  city  bd  gender  registered_via  \
0       268120       3.0    12  35  female               9   
1       265926       3.0    12  35  female               9   
2       217896       3.0    12  35  female               9   
3       197926      52.0     1   0      -1               7   
4       110759      52.0     1   0      -1               7   

   registration_init_time  expiration_date  registration_year  \
0                20140209         20170910               2014   
1                20140209         20170910               2014   
2                20140209         20170910               2014   
3                20160818         20170917               2016   
4                20160818         20170917               2016   

   registration_month  registration_date  
0                   2                  9  
1                   2                  9  
2                   2                  9  
3                   8                 18  
4                   8                 18

<h2>Collection work for members and songs</h2>

In [101]:
import gc
del members,songs;gc.collect();

In [102]:
cols=list(train.columns)
cols.remove('target')

In [103]:
train.head(6)

msno  \
0  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   
1  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
2  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
3  Xumu+NIjS6QYVxDS4/t3SawvJ7viT9hPKXmf0RtLNx8=   
4  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   
5  FGtllVqz18RPiwJj/edr2gV78zirAiY/9SmYvia+kCg=   

                                        song_id source_system_tab  \
0  BBzumQNXUHKdEBOB7mAJuzok+IJA1c2Ryg/yzTF6tik=           explore   
1  bhp/MpSNoqoxOIB+/l8WPqu6jldth4DIpCm3ayXnJqM=        my library   
2  JNWfrrC7zNN7BdMpsISKa4Mw+xVJYNnxXh3/Epw7QgY=        my library   
3  2A87tzfnJTSWqD7gIZHisolhe4DMdzkbd6LzO1KHjNs=        my library   
4  3qm6XTZ6MOCU11x8FIVbAGH5l5uMkT3/ZalWG1oo2Gc=           explore   
5  3Hg5kugV1S0wzEVLAEfqjIV5UHzb7bCrdBRQlGygLvU=           explore   

    source_screen_name      source_type  target      artist_name genre_ids  \
0              Explore  online-playlist       1         Bastille       359   
1  Local playlist more   local-playlist       1  Various Artists      1259   
2  Local playlist more   local-playlist       1              Nas      1259   
3  Local playlist more   local-playlist       1         Soundway      1019   
4              Explore  online-playlist       1      Brett Young      1011   
5              Explore  online-playlist       1        Desiigner      1259   

   song_length  language  city  bd  gender  registered_via  \
0       206471      52.0     1   0      -1               7   
1       284584      52.0    13  24  female               9   
2       225396      52.0    13  24  female               9   
3       255512      -1.0    13  24  female               9   
4       187802      52.0     1   0      -1               7   
5       247803      52.0     1   0      -1               7   

   registration_init_time  expiration_date  registration_year  \
0                20120102         20171005               2012   
1                20110525         20170911               2011   
2                20110525         20170911               2011   
3                20110525         20170911               2011   
4                20120102         20171005               2012   
5                20120102         20171005               2012   

   registration_month  registration_date  
0                   1                  2  
1                   5                 25  
2                   5                 25  
3                   5                 25  
4                   1                  2  
5                   1                  2

In [104]:
test.head()

id                                          msno  \
0   0  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
1   1  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
2   2  RdINwi/2oZyfF8FR5XiSONsD3acLPF31HJ67ZMd1Gt0=   
3   3  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   
4   4  kOXcer+//oOLtOB6JLhxNqj/3YyTH2m+kWfvYTtMc7I=   

                                        song_id source_system_tab  \
0  gpioqmDgSWfpkkCSQts5GMAGaM1TFH5zpVawYOjxIog=        my library   
1  CQ8ouYUSxeEMbNaql4pAL35CqqJKgd0Ms7tH3DmHOJ8=        my library   
2  NA/ZT/ard6XUWrnCvqbEnQLiXnYyP2m4Zq3jWvjLCPY=        my library   
3  wloUHWvwV3EYBJsb0hUG1tIeRlgkrf8QzkootXtY/v0=            search   
4  UZ75pCG2butQs9dVI1s4tedeDZGRwQ551HGdRsVm29Y=            search   

     source_screen_name      source_type       artist_name genre_ids  \
0   Local playlist more    local-library     謝和弦 (R-chord)       465   
1   Local playlist more    local-library     謝和弦 (R-chord)       465   
2   Local playlist more    local-library  王菀之 (Ivana Wong)       458   
3  Online playlist more  online-playlist     Thomas Newman       921   
4  Online playlist more  online-playlist    Marcelo Zarvos       921   

   song_length  language  city  bd  gender  registered_via  \
0       268120       3.0    12  35  female               9   
1       265926       3.0    12  35  female               9   
2       217896       3.0    12  35  female               9   
3       197926      52.0     1   0      -1               7   
4       110759      52.0     1   0      -1               7   

   registration_init_time  expiration_date  registration_year  \
0                20140209         20170910               2014   
1                20140209         20170910               2014   
2                20140209         20170910               2014   
3                20160818         20170917               2016   
4                20160818         20170917               2016   

   registration_month  registration_date  
0                   2                  9  
1                   2                  9  
2                   2                  9  
3                   8                 18  
4                   8                 18

<h2>LabelEncoder</h2>

In [105]:
for col in tqdm(cols):
    if train[col].dtype=='object':
        train[col]=train[col].apply(str)
        test[col]=test[col].apply(str)

        le=LabelEncoder()
        train_values=list(train[col].unique())
        test_values=list(test[col].unique())
        le.fit(train_values+test_values)
        train[col]=le.transform(train[col])
        test[col]=le.transform(test[col])

100%|██████████| 18/18 [00:00<00:00, 374.20it/s]


<h2>Song Popularity</h2>

In [106]:
unique_songs=range(max(train['song_id'].max(),test['song_id'].max()))
song_popularity=pd.DataFrame({'song_id':unique_songs,'popularity':0})
train_sorted=train.sort_values('song_id')
train_sorted.reset_index(drop=True,inplace=True)
test_sorted=test.sort_values('song_id')
test_sorted.reset_index(drop=True,inplace=True)

<h2>Model Training and Prediction</h2>

In [107]:
x=np.array(train.drop(['target'],axis=1))
y=train['target'].values

x_test=np.array(test.drop(['id'],axis=1))
ids=test['id'].values

del train,test; gc.collect();

x_train,x_valid,y_train,y_valid=train_test_split(x,y,test_size=0.1,random_state=12)

del x,y; gc.collect();

d_train=lgb.Dataset(x_train,label=y_train)
d_valid=lgb.Dataset(x_valid,label=y_valid)

watchlist=[d_train,d_valid]


In [109]:
print('Training LGBM model...')

params={}
params['learning_rate']=0.4
params['application']='binary'
params['max_depth']=15
params['num_leaves']=2**8
params['verbosity']=0
params['metric']='auc'

model=lgb.train(params,train_set=d_train,num_boost_round=200,valid_sets=watchlist,\
    early_stopping_rounds=10,verbose_eval=10)



Training LGBM model...
[LightGBM] [Warning] Auto-choosing col-wise multi-threading, the overhead of testing was 0.000869 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
Training until validation scores don't improve for 10 rounds
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[

In [110]:
print('Making predictions and saving them...')
p_test=model.predict(x_test)
p_test

Making predictions and saving them...


array([0.88148306, 0.88047877, 0.90300392, ..., 0.48220867, 0.95464711,
       0.96031486])

<h2>Creating csv file to store id and target columns</h2>

In [112]:
subm=pd.DataFrame()
subm['id']=ids
subm['target']=p_test
subm.to_csv('submission.csv.gz',compression='gzip', index=False, float_format = '%.5f')
print('Zipped..')

Zipped..


In [113]:
yhat=(p_test>0.5).astype(int)
comp=(yhat==ytr).astype(int)
acc=comp.sum()/comp.size*100
print('The accuracy of lgbm model on test data is: {0:f}%'.format(acc))

The accuracy of lgbm model on test data is: 78.800000%


In [115]:
rd_seed=np.random.uniform(0,1,nts)
yhat_rand=(rd_seed>0.5).astype(int)
comp_rand=(yhat_rand==ytr).astype(int)
acc_rand=comp_rand.sum()/comp_rand.size*100
print('The accuracy of random model on test data is :{0:f}%'.format(acc_rand))


The accuracy of random model on test data is :50.066667%


<h2>Conclusion</h2>
<h3>The accuracy of lgbm model is 78.80%</h3>
<h3>The accuracy of random model is 50.06%</h3>

<h2>Hence we conclude that lgbm model is better than random model</h2>